### PR TITLE
Add `transactional` attribute to some call

### LINKED ISSUE #739: Ensure call which interact with internal transaction should be transactional.

### PR DESCRIPTION
**Pull request by AsceticBear**
Close #739

### CODE REVIEW

**Comment by xiaoch05** [comment]
@AsceticBear why add these `transactional` attribute?
we don't need to do that, the call will not change any storage state expect evm state.

**Comment by AsceticBear** [comment]
> the call will not change any storage state expect evm state.

Yes, that's true. However, we need to ensure that the changed storage in other pallets reverts when some error happened in the internal transaction(evm exectuion).

File: frame/bridge/s2s/issuing/src/lib.rs [hunk 1]
```diff
@@ -102,7 +102,7 @@ pub mod pallet {
 		#[pallet::weight(
 			<T as Config>::WeightInfo::asset_burn_event_handle()
 		)]
-		#[frame_support::transactional]
+		#[transactional]
```
Comment (AsceticBear) [inline frame/bridge/s2s/issuing/src/lib.rs:105]: This one is necessary, right? Since we perform a `transfer` before `burn_and_remote_unlock`.

Comment (xiaoch05) [reply frame/bridge/s2s/issuing/src/lib.rs:105]: yes

**Comment by xiaoch05** [comment]
> > the call will not change any storage state expect evm state.
> 
> Yes, that's true. ~However, we need to ensure that the changed storage in other pallets reverts when some error happened in the internal transaction(evm exectuion).~
> 
> I check again these calls, as you say, some are unnecessary. In my view, it would be much safer to add this attribute because these calls always have relations with other pallets, the logic is somehow complicated. Adding `[transactional]` here has no extra bad influence, but can prevent some potential issues in the future.

I can't agree with this. The call's logic is clear. If we add this unnecessary attribute, users will always recheck why transactional attribute here. And do this attribute cause more weight cost?

**Comment by AsceticBear** [comment]
@HackFisher @AurevoirXavier  Any suggestions?

**Comment by HackFisher** [comment]
which kind of dispatch call should be transactional(success or fail without state change)?

This PR LGTM, the dvm internal call will result error if fail right, may be should merge after #713？

**Comment by AsceticBear** [comment]
> which kind of dispatch call should be transactional(success or fail without state change)?

https://substrate.dev/docs/en/knowledgebase/runtime/storage#verify-first-write-last

A example call like this:

```rs
// Pallet A
pub fn call_a1(origin: OriginFor<T>, ...) -> DispatchResultWithPostInfo {
	let user = ensure_signed(origin)?;
        Self::call_a2()?;   // this call change A's runtime storage

        B::call_b()?   // If this call failed and returns, we should ensure that the storage modified in `call_a2` is reverted.
        
	Ok(().into())
}
```

> the dvm internal call will result error if fail, right？ may be should merge after #713？

Internal calls or transacts return an error after 713.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**